src/data/load_geometry.js

```diff
@@ -45,25 +46,24 @@ function pointToLineDist(px, py, ax, ay, bx, by) {
  * @param {VectorTileFeature} feature
  * @private
  */
-export default function loadGeometry(feature: VectorTileFeature, canonical): Array<Array<Point>> {
-
-    const projectionTransform = projection && canonical ? projection.tileTransform(canonical) : null;
-    const z2 = Math.pow(2, canonical.z);
+export default function loadGeometry(feature: VectorTileFeature, canonical?: CanonicalTileID): Array<Array<Point>> {
     const featureExtent = feature.extent;
 
     function reproject(p) {
-        if (projectionTransform) {
+        if (projection && projection.name === 'mercator' || !canonical) {
+            const scale = EXTENT / featureExtent;
+            const p_ = new Point(Math.round(p.x * scale), Math.round(p.y * scale));
+            return clampPoint(p_);
+        } else {
+            const cs = projection.tileTransform(canonical);
+            const z2 = Math.pow(2, canonical.z);
```

Comment:
It's redundant to calculate tile transform for every reprojected point, since it's the same for all points within a feature — we should move this to `loadGeometry` scope.